core/controllers/domain_objects_validator_test.py

```diff
@@ -245,3 +247,63 @@ def test_valid_images_do_not_raises_exception(self) -> None:
             ) as f:
                 files[filename] = f.read()
         domain_objects_validator.validate_suggestion_images(files)
+
+
+class ValidateExplorationChangeTests(test_utils.GenericTestBase):
+    """Tests to validate exploration change coming from frontend"""
+
+    def test_incorrect_exp_state_domain_content(self) -> None:
+        """Tests the state content of exploration change."""
+        incorrect_change_dict = {
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'property_name': exp_domain.STATE_PROPERTY_CONTENT,
+            'state_name': 'New state',
+            'old_value': state_domain.SubtitledHtml(
+                'content', '').to_dict(),
+            'new_value': state_domain.SubtitledHtml(
+                'content',
+                '<oppia-noninteractive-image filepath-with-value='
+                '"&quot;abc.png&quot;" caption-with-value="&quot;'
+                '&quot;" alt-with-value="&quot;&quot;">'
+                '</oppia-noninteractive-image>').to_dict()
+        }
+        with self.assertRaisesRegex(
+            Exception, 'Image tag \'alt-with-value\' attribute '
+            'should not be less than 5.'
+        ):
+            domain_objects_validator.validate_exploration_change(
+                incorrect_change_dict)
+
+    def test_incorrect_exp_state_domain_translations(self):
+        """Tests the state translations of exploration change."""
+        incorrect_change_dict = {
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'property_name': exp_domain.STATE_PROPERTY_WRITTEN_TRANSLATIONS,
+            'state_name': 'New state',
+            'old_value': state_domain.WrittenTranslations({
+                'content': {
+                    'en': state_domain.WrittenTranslation(
+                        'html', '', False)
+                }
+            }).to_dict(),
+            'new_value': state_domain.WrittenTranslations({
+                'content': {
+                    'en': state_domain.WrittenTranslation(
+                        'html',
+                        (
+                            '<oppia-noninteractive-image filepath-with-value='
+                            '"&quot;abc.png&quot;" caption-with-value="&quot;'
+                            '&quot;" alt-with-value="&quot;&quot;">'
+                            '</oppia-noninteractive-image>'
+                        ),
+                        True
+                    )
+                }
+            }).to_dict()
+        }
+        with self.assertRaisesRegex(
+            Exception, 'Image tag \'alt-with-value\' attribute '
```

Comment:
Same note as above.

Also, these tests seem rather light for the added code. Aren't there other failure cases (e.g. validation issues when converting to domain objects)? Think of all the possible behaviours that you want the function to fulfill, and write a test for each of them.

Reply:
Thanks for pointing it out, you are correct. I have added the case for invalid `cmd`. I think now every scenario is taken care of. Please let me know in case I missed something.
Thanks